x/posts/legacy/v0.15.0/migrate_test.go

```diff
@@ -0,0 +1,200 @@
+			},
+		},
+		UsersPollAnswers: []v0150posts.UserAnswersEntry{
+			{
+				PostID: string(postID),
+				UserAnswers: []v0150posts.UserAnswer{
+					{
+						User:    postCreator.String(),
+						Answers: []string{"1", "2"},
+					},
+				},
+			},
+		},
+		PostReactions: []v0150posts.PostReactionsEntry{
+			{
+				PostID: string(postID),
+				Reactions: []v0150posts.PostReaction{
+					{
+						ShortCode: ":fire:",
+						Value:     "🔥",
+						Owner:     postCreator.String(),
+					},
+					{
+						ShortCode: ":my_house:",
+						Value:     "https://myHouse.png",
+						Owner:     postCreator.String(),
+					},
+				},
+			},
+		},
+		RegisteredReactions: []v0150posts.RegisteredReaction{
+			{
+				ShortCode: ":my_house:",
+				Value:     "https://myHouse.png",
+				Subspace:  subspace,
+				Creator:   postCreator.String(),
+			},
+		},
+		Params: v080posts.Params{
+			MaxPostMessageLength:            sdk.NewInt(10),
+			MaxOptionalDataFieldsNumber:     sdk.NewInt(10),
+			MaxOptionalDataFieldValueLength: sdk.NewInt(10),
+		},
+	}
+
+	migrated := v0150posts.Migrate(v0130genesisState)
+
+	// Check for posts
+	require.Len(t, expectedGenState.Posts, len(migrated.Posts))
+	for index, post := range migrated.Posts {
+		require.Equal(t, expectedGenState.Posts[index].PostID, post.PostID)
+		require.Equal(t, expectedGenState.Posts[index].ParentID, post.ParentID)
+		require.Equal(t, expectedGenState.Posts[index].OptionalData, post.OptionalData)
+	}
+
+	// Check for users poll answers
+	require.Len(t, expectedGenState.UsersPollAnswers, len(migrated.UsersPollAnswers))
+	for index, userAnswersEntry := range migrated.UsersPollAnswers {
+		require.Equal(t, expectedGenState.UsersPollAnswers[index].PostID, userAnswersEntry.PostID)
+		for idx, answers := range userAnswersEntry.UserAnswers {
+			require.Equal(t, expectedGenState.UsersPollAnswers[index].UserAnswers[idx], answers)
+		}
+	}
+
+	// Check for post reactions
+	require.Len(t, expectedGenState.PostReactions, len(migrated.PostReactions))
+	for index, postReactionEntry := range migrated.PostReactions {
+		require.Equal(t, expectedGenState.PostReactions[index].PostID, postReactionEntry.PostID)
+		for idx, postReaction := range postReactionEntry.Reactions {
+			require.Equal(t, expectedGenState.PostReactions[index].Reactions[idx], postReaction)
+		}
+	}
+
+	// Check for registered reactions
+	require.Len(t, expectedGenState.RegisteredReactions, len(migrated.RegisteredReactions))
+	for index, regReaction := range migrated.RegisteredReactions {
+		require.Equal(t, expectedGenState.RegisteredReactions[index].Creator, regReaction.Creator)
+		require.Equal(t, expectedGenState.RegisteredReactions[index].ShortCode, regReaction.ShortCode)
+		require.Equal(t, expectedGenState.RegisteredReactions[index].Subspace, regReaction.Subspace)
+		require.Equal(t, expectedGenState.RegisteredReactions[index].Value, regReaction.Value)
```

Comment:
I would suggest allocating a variable to simplify the checks here

```suggestion
		expected := expectedGenState.RegisteredReactions[index]
		require.Equal(t, expected.Creator, regReaction.Creator)
		require.Equal(t, expected.ShortCode, regReaction.ShortCode)
		require.Equal(t, expected.Subspace, regReaction.Subspace)
		require.Equal(t, expected.Value, regReaction.Value)
```